lib/protobuf/json/encode.ex

```diff
@@ -0,0 +1,139 @@
+defmodule Protobuf.JSON.Encode do
+  @moduledoc """
+  Internal JSON encoding API.
+  """
```

Comment:
Let's make this module `@moduledoc false` and document the option types in the `Protobuf.JSON` module.